subprojects/packagefiles/pugixml/meson.build

```diff
@@ -0,0 +1,61 @@
+project('pugixml', 'cpp', version: '1.11', meson_version: '>=0.46.0')
+
+cxx = meson.get_compiler('cpp')
+
+sources = files('src/pugixml.cpp')
+
+if host_machine.system() == 'windows'
+  win_mod = import('windows')
+  sources += win_mod.compile_resources('scripts/pugixml_dll.rc')
+endif
+
+args = []
+if cxx.get_id() == 'msvc'
+  if get_option('default_library') == 'shared'
+    args +='-DPUGIXML_API=__declspec(dllexport)'
+  elif get_option('default_library') == 'both'
+    error('cannot build both shared and static on MSVC due to declspec')
+  endif
+endif
+
+foreach opt: ['wchar_mode', 'compact', 'no_xpath', 'no_stl', 'no_exceptions']
+  if get_option(opt)
+    args += '-DPUGIXML_' + opt.to_upper()
+  endif
+endforeach
+
+pugixml_lib = library('pugixml', sources,
+  cpp_args: args,
+  version: meson.project_version(),
```

Comment:
Nitpicking: meson doesn't set that by default?

Reply:
No it does not, this is a soname ABI. A pkg-config version is set by default, though.